### PR TITLE
Copy map argument in extraVars

### PR DESCRIPTION
**Pull request by moolitayer**
See https://github.com/moolitayer/awx-client-go/pull/19#discussion_r186282175

### CODE REVIEW

**Comment by moolitayer** [comment]
@zgalor do we need a deep copy?

File: awx/job_template_launch_resource.go [hunk 1]
```diff
@@ -84,7 +84,11 @@ type JobTemplateLaunchPostRequest struct {
 
 // ExtraVars set a map or external variables sent to the AWX job.
 func (r *JobTemplateLaunchPostRequest) ExtraVars(value map[string]interface{}) *JobTemplateLaunchPostRequest {
-	r.extraVars = value
+	extraVars := make(map[string]int)
+	for key, value := range value {
+		extraVars[key] = value
+	}
+	r.extraVars = extraVars
```
Comment (cben) [inline awx/job_template_launch_resource.go:91]: what's the desired semantics if you call ExtraVars after having called ExtraVar a few times?
should it append only or replace?

Comment (moolitayer) [reply awx/job_template_launch_resource.go:91]: Good point. Not clear from the semantics. Two thoughts
- If we revise the current behavior to always append we need to provide a way to clear existing values
- Maybe we can use setters as described [here](https://golang.org/doc/effective_go.html#Getters) and then we can have SetExtraVars that replaces and AddExtraVar that appends

Open for better suggestions...

File: awx/job_template_launch_resource.go [hunk 2]
```diff
@@ -84,7 +84,11 @@ type JobTemplateLaunchPostRequest struct {
 
 // ExtraVars set a map or external variables sent to the AWX job.
 func (r *JobTemplateLaunchPostRequest) ExtraVars(value map[string]interface{}) *JobTemplateLaunchPostRequest {
-	r.extraVars = value
+	extraVars := make(map[string]int)
```
Comment (zgalor) [inline awx/job_template_launch_resource.go:87]: `make(map[string]interface{})`

**Comment by zgalor** [comment]
if we want to support `interface{}` (as we should), we should probably deepcopy it.
See https://github.com/openshift/autoheal/blob/master/pkg/apis/autoheal/types.go#L59-L75

**Comment by jhernand** [comment]
Requiring deep copy will probably be a problem for other users of the library. I'd suggest to avoid it. I think it is OK if you clearly document it.

**Pull request closed without merging**